shield/internal/parser/parser_test.go

```diff
@@ -96,3 +96,49 @@ func TestBooleanOperations(t *testing.T) {
 	require.True(t, ok, "expression is not *ast.InfixExpression. got=%T", and.Left)
 	require.Equal(t, "&&", and.Operator)
 }
+
+func TestParser(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{
+			`5`,
+			"5",
+		},
+		{
+			`true`,
+			"true",
+		},
+		{
+			`false`,
+			"false",
+		},
+		{
+			`warden1234`,
+			"warden1234",
+		},
+		{
+			`true || true && false`,
+			"(true || (true && false))",
+		},
+		{
+			`(true && false)`,
+			"(true && false)",
+		},
+		{
+			`true || (true && false)`,
+			"(true || (true && false))",
+		},
```

Comment:
Hey @Pitasi I suggest to add case with built-in function call, something like this

```
{
  `((true || (address) || ((any(2, true) || false) && true)))`,
  "((true || address) || ((any(2, true) || false) && true))",
}
```
It will require a little change is Stringify, I believe it can be done as Stringify is not used anywhere except tests
```
case *Expression_CallExpression:
		return fmt.Sprintf("%s(%s)", n.CallExpression.Function.Value, stringifyExpressions(n.CallExpression.Arguments))
```

Reply:
Good catch, thanks, it was another bug inside the Stringify function. I have fixed it now in [d91da01](https://github.com/warden-protocol/wardenprotocol/pull/225/commits/d91da0158ce4a068648bc1254543dcd48ee892e5) and added the new test case.